intelligence_asn.go

```diff
@@ -0,0 +1,104 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+// ErrMissingASN is for when ASN is required but not set.
+var ErrMissingASN = errors.New("required asn missing")
+
+// ASNInfo represents ASN information.
+type ASNInfo struct {
+	ASN         int      `json:"asn"`
+	Description string   `json:"description"`
+	Country     string   `json:"country"`
+	Type        string   `json:"type"`
+	DomainCount int      `json:"domain_count"`
+	TopDomains  []string `json:"top_domains"`
+}
+
+// IntelligenceASNOverviewParameters represents parameters for an ASN request.
+type IntelligenceASNOverviewParameters struct {
+	AccountID string
+	ASN       int
+}
+
+// IntelligenceASNResponse represents an API response for ASN info.
+type IntelligenceASNResponse struct {
+	Response
+	Result []ASNInfo `json:"result,omitempty"`
+}
+
+// IntelligenceASNSubnetsParameters represents parameters for an ASN subnet request.
+type IntelligenceASNSubnetsParameters struct {
+	AccountID string
+	ASN       int
+}
+
+// IntelligenceASNSubnetResponse represents an ASN subnet API response.
+type IntelligenceASNSubnetResponse struct {
+	ASN          int      `json:"asn,omitempty"`
+	IPCountTotal int      `json:"ip_count_total,omitempty"`
+	Subnets      []string `json:"subnets,omitempty"`
+	Count        int      `json:"count,omitempty"`
+	Page         int      `json:"page,omitempty"`
+	PerPage      int      `json:"per_page,omitempty"`
+}
+
+// IntelligenceASNOverview get overview for an ASN number
+//
+// API Reference: https://api.cloudflare.com/#asn-intelligence-get-asn-overview
+func (api *API) IntelligenceASNOverview(ctx context.Context, params IntelligenceASNOverviewParameters) ([]ASNInfo, error) {
+	// Make sure Account ID is set
```

Comment:
let's do without these style of comments. this [blog from stackoverlow](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/) has a good write up of why comments like this are something to avoid and what to do about it if you think they are nessecary.